build.roc

```diff
@@ -0,0 +1,152 @@
+    weaver: "https://github.com/smores56/weaver/releases/download/0.2.0/BBDPvzgGrYp-AhIDw0qmwxT0pWZIQP_7KOrUrZfp_xw.tar.br",
+}
+
+import cli.Task exposing [Task]
+import cli.Cmd
+import cli.Stdout
+import cli.Arg
+import weaver.Opt
+import weaver.Cli
+
+RocTarget : [
+    MacosArm64,
+    MacosX64,
+    LinuxArm64,
+    LinuxX64,
+    WindowsArm64,
+    WindowsX64,
+]
+
+main =
+
+    cliParser =
+        Cli.weave {
+            release: <- Opt.flag { short: "r", help: "Release build" },
+        }
+        |> Cli.finish {
+            name: "basic-webserver",
+            version: "",
+            authors: ["Luke Boswell <https://github.com/lukewilliamboswell>"],
+            description: "This build script generates the binaries and packages the platform for distribution.",
+        }
+        |> Cli.assertValid
+
+    when Cli.parseOrDisplayMessage cliParser (Arg.list!) is
+        Ok args -> run args
+        Err message -> Task.err (Exit 1 message)
+
+run = \{ release } ->
+
+    # generate glue for builtins
+    printInfoLine! "generating glue for builtins"
+    Cmd.exec "roc" ["glue", "glue.roc", "crates/", "platform/main.roc"]
+    |> Task.mapErr! ErrGeneratingGlue
+
+    # get the roc target for the native machine
+    target = getNativeTarget!
+
+    # build the app stub shared library (for processing surgical linker binaries)
+    appStubBuildPath = "platform/libapp.$(appStubExt target)"
+    printInfoLine! "roc build the app stub shared library at $(appStubBuildPath)"
+    Cmd.exec "roc" ["build", "--lib", "platform/libapp.roc"]
+    |> Task.mapErr! ErrBuildingAppStub
+
+    # build the roc host binaries
+    (cargoBuildArgs, message) =
+        if release then
+            (["build", "--release"], "building roc host binaries in release mode")
+        else
+            (["build"], "building roc host binaries in debug mode")
+
+    printInfoLine! message
+    Cmd.exec "cargo" cargoBuildArgs
+    |> Task.mapErr! ErrBuildingRocHostBinaries
+
+    # move the prebuilt binary to the platform directory
+    prebuiltLegacyBinaryBuildPath =
+        if release then
+            "target/release/libhost.a"
+        else
+            "target/debug/libhost.a"
+
+    prebuiltLegacyBinaryPath = "platform/$(prebuiltBinaryName target)"
+    printInfoLine! "moving the prebuilt binary from $(prebuiltLegacyBinaryBuildPath) to $(prebuiltLegacyBinaryPath)"
+    Cmd.exec "cp" [prebuiltLegacyBinaryBuildPath, prebuiltLegacyBinaryPath]
+    |> Task.mapErr! ErrMovingPrebuiltLegacyBinary
+
+    #SURGICAL LINKER IS NOT YET SUPPORTED need to merge https://github.com/roc-lang/roc/pull/6808
+    #prebuiltSurgicalBinaryBuildPath =
+    #    if release then
+    #        "target/release/host"
+    #    else
+    #        "target/debug/host"
+    #Cmd.exec "roc" ["preprocess-host", prebuiltSurgicalBinaryBuildPath, "platform/main.roc", "platform/libapp.$(appStubExt target)"]
+    #|> Task.mapErr! ErrPreprocessingSurgicalBinary
```

Comment:
@Anton-4 -- if you are looking at https://github.com/roc-lang/roc/pull/6808 you can uncomment these out and it should work.

Reply:
6808 is merged. 

We can make this available when the new nightly is out I guess.